main/network.c

```diff
@@ -859,54 +859,52 @@ php_socket_t php_network_connect_socket_to_host(const char *host, unsigned short
 		if (sa) {
 			/* make a connection attempt */
 
-			if (bindto) {
-				struct sockaddr *local_address = NULL;
-				int local_address_len = 0;
+			while (bindto) {
+				struct {
+					int len;
```

Comment:
It looks like `len` is not actually used, beyond checking that something matched. I'd make it a bool flag or change control flow to not need it.

Reply:
Not sure what I was thinking here, len is used in the bind call...

Reply:
len is used by bind function for the last param, I do not understand how can I make it be bool flag, I applied the other suggestion, does it look better now?